types/mongodb/index.d.ts

```diff
@@ -936,10 +937,8 @@ type EnhancedOmit<T, K> =
     Omit<T, K>;
 
 type ExtractIdType<TSchema> =
-    TSchema extends { _id: infer U } // user has defined a type for _id
-    ? {} extends U ? Exclude<U, {}> :
-      unknown extends U ? ObjectId : U
-    : ObjectId; // user has not defined _id on schema
+    TSchema extends { _id: any } // user has defined a type for _id
+    ? TSchema['_id'] : ObjectId;
```

Comment:
The new version is not equivalent to the old one. `{ _id: any }` or  `{ _id: object}` returned `ObjectId`

It was originally added like this by @HosseinAgha

Reply:
I suppose that's true; however, I would argue that if _id has a specific value then it should always use that value.

If you are going to argue that any should always be treated as a ObjectId for this then `any` should be treated the same as `{_id: ObjectId}` but in that case it would return `any`.

Again, why is the correct thing here to ignore specified types?

Reply:
I think that maybe this can be another issue / pull request, focused on this.